### DISCUSSION TITLE
:bug: [Bug Report] Emojis not Shown Correctly in KDE Konsole with Default Settings

### DISCUSSION THREAD

**Discussion by kevinmatthes** [ttytm/wthrr-the-weathercrab#80] [Q&A] [answered]
First of all, thank you for this amazing weather CLI!

When running Wthrr in KDE Konsole (v20.12.3) with its default configuration (font Hack, 10 pt) on Linux, I noticed that the emojis are not shown appropriately.  That is no major problem as the given data is also understandable without having the emojis but in some situations, the user experience might suffer from it.  For Konsole, the default font can be changed but in terminal environments where this is not possible, the results would not be shown appropriately, especially regarding the hourly forecast.

Which are the recommended fonts to use with Wthrr?  Is there a list of them in the documentation?

**Comment by ttytm** [comment]
Thanks for reaching out @kevinmatthes!
Nice to to hear that you like it 🦀😊.

Yep, a font that supports those icons is a requirement to use the app unimpaired. Any Nerd Font would do. And if it's not already installed by default on your system, a Unicode font core package like Noto might be necessary. They are listed in the requirements https://github.com/tobealive/wthrr-the-weathercrab/blob/main/INSTALL.md#requirements. 

The only tiny issue I'm having with konsole is the size for the precipitation probability icon. Which might be change with the v3.0 nerd-fonts release.

**Comment by kevinmatthes** [comment]
I already tried to change to Noto Mono regular but the result still shows either tofu or whatever might be mapped to the character code, for instance media player symbols for weather conditions.

Is there any official Debian installation package for the Nerd fonts available?

**Comment by ttytm** [comment, marked as answer]
It's really a nerd font that needs to be set. Unfortunately I think there is no way around it. For the availability of some characters noto is funnily still necessary, even though it's not set as font. 

The last time I checked on a deb based system there were no nerd font packages available via apt. 
The nerd-fonts repo holds variants for most monospace fonts, including hack https://github.com/ryanoasis/nerd-fonts/tree/master/patched-fonts.

If you prefer a short script that can to easily pull fonts. Here is one from my dotfiles. It requires `subversion` to be installed to pull those parts from upstream nerd-fonts repo. Adapt the JetBrains / Fantasque urls accordingly to you desired font.

<details>
<summary><code>install_fonts.sh</code></summary>

```sh
#!/bin/bash

FONT_DIR="$HOME/.local/share/fonts"

if [ ! -d "$FONT_DIR" ]; then
	mkdir "$FONT_DIR"
fi

cd "$FONT_DIR"

# Nerd Fonts
svn export https://github.com/ryanoasis/nerd-fonts/trunk/patched-fonts/JetBrainsMono/Ligatures JetBrainsNFM 
svn export https://github.com/ryanoasis/nerd-fonts/trunk/patched-fonts/FantasqueSansMono FantasqueSansNFM

# Regular Fonts
mkdir FantasqueSansMono && cd FantasqueSansMono \
	&& curl -LO https://github.com/belluzj/fantasque-sans/releases/latest/download/FantasqueSansMono-LargeLineHeight.tar.gz \
	&& tar -zxf FantasqueSansMono-LargeLineHeight.tar.gz TTF \
	&& mv TTF/*.ttf ./ \
	&& rm -r TTF && rm FantasqueSansMono-LargeLineHeight.tar.gz \

fc-cache -fv

exit 0
```

</details>

**Reply by kevinmatthes** [reply]
Thank you, this answers the question!

**Comment by ttytm** [comment]
I'll move this to the questions section, as what we are facing is not a bug that can be fixed in a regular fashion. Let me know if I can help further!
